### PR TITLE
Added api authorization, fixed warnings

### LINKED ISSUE #53: Add Api Authorization

### PR DESCRIPTION
**Pull request by guillemurano**
closes #53

### CODE REVIEW

File: src/DonateTo.WebApi/Middlewares/ExceptionHandlerMiddleware.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public async Task InvokeAsync(HttpContext httpContext)
             }
             catch (Exception exc)
             {
-                Logger.Error("Unhandled Exception occurred",exc);
+                Logger.Error("Unhandled Exception occurred", exc);
```
Comment (nicolasgranata) [inline src/DonateTo.WebApi/Middlewares/ExceptionHandlerMiddleware.cs:34]: Undo the changes on this class middleware because we changed it to another approach.

**Comment by nicolasgranata** [comment]
Please resolve the merge conflicts

File: src/DonateTo.Services/Services/UserService.cs [hunk 2]
```diff
@@ -18,22 +19,22 @@ public class UserService : IUserService
             _userRepository = userRepository;
         }
 
-        public User Create(User user)
+        public User Create(User entity)
```
Comment (nicolasgranata) [inline src/DonateTo.Services/Services/UserService.cs:22]: Should this class inherit from BaseService for CRUD operations and use the IUserService interface for specialization?

Comment (guillemurano) [reply src/DonateTo.Services/Services/UserService.cs:22]: It cannot implement BaseService because User is implementing IdentityUser<long> instead of EntityBase needed for BaseService

File: src/DonateTo.IdentityServer/appsettings.json [hunk 3]
```diff
@@ -9,5 +9,47 @@
...
+    "Clients": [
+      {
+        "Enabled": true,
+        "ClientId": "DonateTo.WebAplication",
+        "ClientName": "DonateTo Web",
+        "ClientSecrets": [
+          {
+            "Value": "0a99e6ba3dddf18fbe47973a65da3e5d8ae4f5a689951d47602c108ca252c3f3"
+          }
+        ], //this is not a test
```
Comment (pablo-bergonzi) [inline src/DonateTo.IdentityServer/appsettings.json:47]: Do we need that comment?

File: src/DonateTo.Services/Services/BaseService.cs [hunk 4]
```diff
@@ -2,86 +2,120 @@
...
 using System;
 using System.Collections.Generic;
+using System.Linq.Expressions;
 
 namespace DonateTo.Services
 {
     public abstract class BaseService<TEntity>: IBaseService<TEntity> where TEntity: EntityBase
     {
         private readonly IRepository<TEntity> _entityRequestRepository;
+        private const string _nullEntityException = "Entity is null.";
```
Comment (pablo-bergonzi) [inline src/DonateTo.Services/Services/BaseService.cs:15]: If we are going to do this, couldn't we go ahead and use custom exceptions?

Comment (guillemurano) [reply src/DonateTo.Services/Services/BaseService.cs:15]: The best approach would be adding a resources file to address all messages, but being an service error, I don't think is necessary to localize that
